maven-api-impl/src/main/java/org/apache/maven/internal/impl/resolver/LocalSnapshotMetadata.java

```diff
@@ -113,6 +113,11 @@ protected void merge(Metadata recessive) {
 
         metadata = metadata.withVersioning(metadata.getVersioning().withSnapshotVersions(versions.values()));
 
+        // just carry-on as-is
+        if (recessive.getPlugins() != null && !recessive.getPlugins().isEmpty()) {
+            metadata = metadata.withPlugins(new ArrayList<>(recessive.getPlugins()));
```

Comment:
should we also check if `metadata` already contains a `plugins`?

Reply:
The `metadata` cannot contain plugins, see how it is created:
* in ctor
* only "related" stuff filled in (Versions)
* point is that "not my stuff" should be just carried over as is